### PR TITLE
Add redux for react application

### PR DESCRIPTION
**Pull request by abdallahmalima**
## In this part I:
- Add Actions and reducer for books
- Add Actions and reducer for categories
- Combine all reducers in the configureStore.js file

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { combineReducers } from 'redux';
```
Comment (FrederickMih) [inline src/redux/configureStore.js]: - [ ] Great job installing reduxjs/toolkit. Now let's kindly use it in this project. Based on the requirements, we are told to use it.

`Imports configureStore  from @reduxjs/toolkit.`

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+import books from './books/books';
+import categories from './categories/categories';
+
+const configureStore = () => combineReducers({ books, categories });
```
Comment (FrederickMih) [inline src/redux/configureStore.js]: -  Great job for using configureStore to create reducers :+1: but you didn't import it. Let's kindly check the comment above :arrow_up:

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+const ADD_BOOK = 'book-store/books/ADD_BOOK';
+const REMOVE_BOOK = 'book-store/books/REMOVE_BOOK';
```
Comment (FrederickMih) [inline src/redux/books/books.js:2]: - Fantastic job for using duck pattern for each value of action type :+1:

File: src/redux/categories/categories.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (FrederickMih) [inline src/redux/categories/categories.js]: - [ ] Let's kindly follow the ducks pattern. You have done it for books and I think you can do it again here :+1: :smile: 

- Following the ducks pattern enables us to easily scale/debug our project in the future.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: @abdallahmalima, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


Screenshot from 2022-10-20 09-38-51


## Highlights :green_circle: 

- Reducers have been correctly created and exported :+1: 
- PR is descriptive :+1: 
- All Linter checks passed :heavy_check_mark: 


## Required Changes ♻️
N/A
_Check the comments under the review._

## Optional suggestions
N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**